### ISSUE TITLE
fix: redo incorrectly does undo on Mac

### ISSUE THREAD

**Issue by tangjeff0** [athensresearch/athens#1004]
**Problem**: When user tries to redo, they actually undo, losing their latest changes!

**Screenshots/Demo**

**Athens Version**

**Comment by kamalmarhubi** [comment]
This is happening on Linux as well, at least in v1.0.0-beta.76.

**Comment by tangjeff0** [comment]
Not sure how to repro this, and when this happens. Are you using a different keyboard @kamalmarhubi ?

**Comment by ssjoleary** [comment]
Ah, I stumbled across this!

Undo/redo gets a bit confused in a couple of scenarios:
* Either undoing or redoing in an empty block deletes that block and the block immediately above it
* Navigating to a different block and either undoing or redoing straight away has a kind of unpredictable behaviour but blocks, other than the block in focus, will start disappearing

Once the bug starts the undo/redo functionality will continue to remove things until everything's gone from the page or until the user starts typing again.

I _think_ the changes are still available in the history, it's the logic for stacking/reapplying the changes that's getting confused (rather than the undo/redo removing it from the history completely). Either way a panicked user that encounters the bug and tries to alternate between undo's and redo's will end up removing things that can't be brought back.

The undo/redo functionality looks to have been updated in March, #808 , and seems nicely contained so I'll take a stab at a fix. I think I've seen @arkRedM about on the Discord too so I'll ping them for some advice 😉
